samples/drivers/touch/src/main.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sample, LOG_LEVEL_INF);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/drivers/display.h>
+
+#define CROSS_WIDTH  20
+#define CROSS_HEIGHT 20
+#define BUFFER_SIZE  ((CROSS_WIDTH) * (CROSS_HEIGHT) * 4)
+#define REFRESH_RATE 100
+
+#ifndef DT_CHOSEN_zephyr_touch
+#error "Unsupported board: zephyr,touch is not assigned"
+#endif
+
+#ifndef DT_CHOSEN_zephyr_display
+#error "Unsupported board: zephyr,display is not assigned"
+#endif
+
+static const struct device *const display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+static const struct device *const touch_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_touch));
+static struct display_capabilities capabilities;
+static struct display_buffer_descriptor buf_desc = {.buf_size = BUFFER_SIZE,
+						    .pitch = CROSS_WIDTH,
+						    .width = CROSS_WIDTH,
+						    .height = CROSS_HEIGHT};
+
+static uint8_t buffer_cross[BUFFER_SIZE] = {0};
+static const uint8_t buffer_cross_empty[BUFFER_SIZE] = {0};
+static struct k_sem sync_x, sync_y;
+
+static struct {
+	size_t x;
+	size_t y;
+} touch_point, touch_point_drawn;
+
+static void touch_event_callback(struct input_event *evt, void *user_data)
+{
+	if (evt->code == INPUT_ABS_X) {
+		touch_point.x = evt->value;
+		k_sem_give(&sync_x);
+	}
+	if (evt->code == INPUT_ABS_Y) {
+		touch_point.y = evt->value;
+		k_sem_give(&sync_y);
+	}
```

Comment:
that's not quite the right way of using the API, you should cache x y locally and generate the event once you see an event with the sync bit set, somewhat like this https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/kscan/kscan_input.c#L26-L50, then you only need some semaphore

Reply:
Thanks for the tip, should be fixed now